src/parser/pipelines/collector.rs

```diff
@@ -281,6 +281,10 @@ impl<'a> Collector<'a> {
                             bytes.next();
                             try_add_item!(JobKind::And);
                         }
+                        Some(&(_, b'!')) => {
+                            bytes.next();
+                            try_add_item!(JobKind::Disown);
```

Comment:
Would it be better to somehow use the existing JobKind with a boolean or something, so it doesn't have to be checked against both Disown and Background every time?

Reply:
It's fine to keep it as it's own variant. There are ways that we could optimize it this way using bitwise operations.